### PR TITLE
e2e reboot: print status and logs for not running/ready pods

### LINKED ISSUE #20099: Collect logs of failed containers in kube-system namespace
We are running heapster, dns server, kube-proxy etc. in kube-system namespace. Some e2e tests try to make sure all those kube-system pods are at ready state before firing the real tests, others just simply ingore them. But both cases might cause e2e flakiness when some of those kube-system containers are in crashloop, for example, https://github.com/kubernetes/kubernetes/issues/19986#issuecomment-174089670 for heapster crashloop, https://github.com/kubernetes/kubernetes/issues/19189#issuecomment-173419306 for influxdb crashloop, etc. When such cases happen, the developer doesn't have much information on why since the nodes might be teared down. 

We should watch those kube-system pods, and onContainerFailure, we should collect the related logs. One simple way is using kubectl logs <Pod> <Container> --previous

cc/ @kubernetes/goog-node @kubernetes/goog-testing

### PR DESCRIPTION
**Pull request by yujuhong**
This fixes #20099

### CODE REVIEW

File: test/e2e/downwardapi_volume.go [hunk 1]
```diff
@@ -32,18 +30,6 @@ import (
 // How long to wait for a log pod to be displayed
 const podLogTimeout = 45 * time.Second
 
-// utility function for gomega Eventually
-func getPodLogs(c *client.Client, namespace, podName, containerName string) (string, error) {
```
Comment (dchen1107) [inline test/e2e/downwardapi_volume.go]: No one is using this utility function?

**Comment by dchen1107** [comment]
LGTM

I am going to merge this one once the build is green. This one can help us to debug why heapster has crashloop, which causes reboot test / gke-ci failure.

File: test/e2e/downwardapi_volume.go (hunk 1 above)
Comment (dchen1107) [inline test/e2e/downwardapi_volume.go]: Ignore my above comment. I thought it is test/e2e/util.go. I saw you moved it to the right file below.

**Comment by piosz** [comment]
Good job @yujuhong. This will be very helpful.

**Pull request merged**